apps/examples/wakaama_client/wakaama_client.c

```diff
@@ -394,6 +314,41 @@ int wakaamaclient_main(int argc, char *argv[])
 			printf("Read a resource value\n");
 			return 0;
 
+		case 'f': {
+			int i;
+			int wifi_count;
+			while (1) {
+				wifi_count = 0;
+				if (dm_conn_wifi_scan() == DM_ERROR_NONE) {
+					printf("DM scan network PASS\n");
+				} else {
+					printf("Failed on wifi scan\n");
+				}
+				sleep(10);
+				if (dm_conn_get_scan_result(&appwifiscan_result) ==  DM_ERROR_NONE) {
+					wifi_iterator = appwifiscan_result;
+					while (wifi_iterator != NULL) {
+						for (i = 0; i < 11; i++) {
+							if (!strncmp(wifi_iterator->ssid, wifi_aps[i], strlen(wifi_aps[i]))) {
+								printf("AP:%d:%s:%s:%d\n", appscan_iteration,
+									   wifi_iterator->ssid,
+									   wifi_iterator->bssid,
+									   wifi_iterator->rssi);
+							}
+						}
+						wifi_count++;
+						wifi_iterator = wifi_iterator->next;
+					}
+				} else {
+					printf("DM_ERROR_NO_DATA\n");
+				}
+				printf("AP:%d:totalAP#:%d\n", appscan_iteration, wifi_count);
+				dm_conn_free_scan_result(&appwifiscan_result);
```

Comment:
It follows from having to scan, and more or less replicates WiFi code at supplicant level. Also, if the user wishes to get data from a NULL pointer argument, (see dm_conn_get_scan_result), then it is the user's responsibility to decide when to free it.